examples/mhp/stencil-1d.cpp

```diff
@@ -50,14 +42,19 @@ int stencil(auto n, auto steps) {
   dr::mhp::iota(a, 100);
   dr::mhp::fill(b, 0);
 
-  auto in = rng::subrange(a.begin() + 1, a.end() - 1);
-  auto out = rng::subrange(b.begin() + 1, b.end() - 1);
+  auto in = rng::ref_view(a);
+  auto out = rng::ref_view(b);
+
   for (std::size_t s = 0; s < steps; s++) {
-    dr::mhp::halo(in).exchange();
-    dr::mhp::transform(in, out.begin(), stencil_op);
+    dr::mhp::halo(in.base()).exchange();
```

Comment:
`base()` is already not needed in this code - removed

Making `dr::mhp::halo(dr::mhp::views::sliding(in)).exchange()` is not obvious how to me how to do it in the cleanest way now. I've left a TODO. If you think this is important I'll create task for it or I'll think and propose some solution in this PR. 

Is it important to do / important to do before merging this PR?

Reply:
Sliding is for stencil and stencil needs halo exchange. But it can be a separate commit. We really need a universal way to get a handle to the underlying container that dr::mhp::halo() and other CPO's can use, instead of trying to handle each CPO individually.